doc/man3/ECDSA_SIG_new.pod

```diff
@@ -124,74 +57,66 @@ ECDSA_SIG_set0() returns 1 on success or 0 on failure.
 ECDSA_SIG_get0_r() and ECDSA_SIG_get0_s() return the corresponding value,
 or NULL if it is unset.
 
-ECDSA_size() returns the maximum length signature or 0 on error.
-
-ECDSA_sign(), ECDSA_sign_ex() and ECDSA_sign_setup() return 1 if successful
-or 0 on error.
-
-ECDSA_do_sign() and ECDSA_do_sign_ex() return a pointer to an allocated
-B<ECDSA_SIG> structure or NULL on error.
-
-ECDSA_verify() and ECDSA_do_verify() return 1 for a valid
-signature, 0 for an invalid signature and -1 on error.
-The error codes can be obtained by L<ERR_get_error(3)>.
-
 =head1 EXAMPLES
 
-Creating an ECDSA signature of a given SHA-256 hash value using the
-named curve prime256v1 (aka P-256).
-
-First step: create an EC_KEY object (note: this part is B<not> ECDSA
-specific)
+Extract signature 'r' and 's' values from a ECDSA 'signature'
+of size 'signaturelen':
 
- int ret;
- ECDSA_SIG *sig;
- EC_KEY *eckey;
+ ECDSA_SIG *obj;
+ const BIGNUM *r, *s;
 
- eckey = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1);
- if (eckey == NULL)
-     /* error */
- if (EC_KEY_generate_key(eckey) == 0)
+ /* Load a signature into the ECDSA_SIG object */
+ obj = d2i_ECDSA_SIG(NULL, &signature, signaturelen);
+ if (obj == NULL)
      /* error */
 
-Second step: compute the ECDSA signature of a SHA-256 hash value
-using ECDSA_do_sign():
-
- sig = ECDSA_do_sign(digest, 32, eckey);
- if (sig == NULL)
+ r = ECDSA_SIG_get0_r(obj);
+ s = ECDSA_SIG_get0_s(obj);
+ if (r == NULL || s == NULL)
      /* error */
 
-or using ECDSA_sign():
+ /* Use BN_bn2binpad() here to convert to r and s into byte arrays */
 
- unsigned char *buffer, *pp;
- int buf_len;
+ ECDSA_SIG_free(obj);
```

Comment:
You're freeing the object that owns the r and s values. Perhaps add a comment that they should not be used from here on?